hypershift-operator/controllers/nodepool/metrics/metrics.go

```diff
@@ -0,0 +1,46 @@
+package metrics
+
+import (
+	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	labelNames = []string{"namespace", "name", "platform"}
+
+	nodePoolSizeMetricName = "hypershift_nodepools_size"
+	nodePoolSize           = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Help: "Number of desired replicas associated with a given NodePool",
+		Name: nodePoolSizeMetricName,
+	}, labelNames)
+
```

Comment:
What's the value of having these two?
I think it'd good to discuss and articulate the SRE story that would leverage and maybe set alert on this for e.g missmatch between both values for x duration before introducing more metrics